### ISSUE TITLE
Autograd compatibility

### ISSUE THREAD

**Issue by andres-fr** [andres-fr/skerch#12]
Currently, PyTorch is being used due to its device-agnostic treatment of tensors.
But autograd-integration should also be possible, and a great perk of having a PyTorch backend.

This has not been explored so far. Since we deal with linear operators, the relation between forward and backward operations is trivial (Jacobian of  matrix is the matrix itself).

So one possible way to implement and test this would be to create a wrapper that takes a matrix or a skerch composite linop and returns an autograd-compatible [torch.autograd.Function](https://docs.pytorch.org/docs/main/autograd.html#torch.autograd.Function) with the same forward behaviour and the corresponding gradients.

Unit tests would require to test this in combination with autograd, perhaps running `torch.backward` and inspecting that returned gradients for wrapped objects indeed coincide with the expected values.

Integration tests could involve replacing e.g a torch SVD with `ssvd` to train a PCA via least-squares and observe similar convergence behaviour.
